protocols/rendezvous/tests/rendezvous.rs

```diff
@@ -79,6 +81,23 @@ async fn given_successful_registration_then_successful_discovery() {
     }
 }
 
+#[tokio::test]
+async fn should_return_error_when_no_external_addresses() {
+    let _ = env_logger::try_init();
+    let namespace = rendezvous::Namespace::from_static("some-namespace");
+    let server = new_server(rendezvous::server::Config::default()).await;
+    let mut client = Swarm::new_ephemeral(rendezvous::client::Behaviour::new);
+
+    let res = client
+        .behaviour_mut()
+        .register(namespace.clone(), *server.local_peer_id(), None);
+
+    match res {
+        Err(RegisterError::NoExternalAddresses) => {}
+        _ => panic!("Should get the RegisterError::NoExternalAddresses"),
+    }
```

Comment:
Can use `unwrap_err` here and then `assert_eq` on the error.

Reply:
In that case, `RegisterError` should implement according traits.
Is it proper to implement traits in production code only for testing purposes?

Reply:
What traits do we need? `Debug` and `PartialEq` those are fine yeah.

Reply:
> What traits do we need? `Debug` and `PartialEq` those are fine yeah.

```
pub struct SigningError {
    msg: String,
    source: Option<Box<dyn Error + Send + Sync>>,
}
```

```
119 | #[derive(Debug, PartialEq)]
    |                 --------- in this derive macro expansion
...
122 |     source: Option<Box<dyn Error + Send + Sync>>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the derive macro `PartialEq`
```

Reply:
<img width="733" alt="image" src="https://github.com/libp2p/rust-libp2p/assets/25565200/4ccf7abc-31a3-4d20-b448-ae07102af92d">

What is the right workaround?

Reply:
You could implement it manually and compare the pointers but that doesn't really make sense in the grand scheme of things.

I'd suggest we just do `unwrap_err` and use `assert!(matches!(err, RegisterError::...))`